modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/repository_entry_browser/js/ItemSelectorRepositoryEntryBrowser.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {delegate} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+
+import ItemSelectorPreview from '../../item_selector_preview/js/ItemSelectorPreview.es';
+import SingleFileUploader from '../../item_selector_uploader/js/SingleFileUploader';
+
+export default function ItemSelectorRepositoryEntryBrowser({
+	closeCaption,
+	itemSelectedEventName,
+	portletNamespace,
+	rootNode,
+	uploaderEnabled = true,
+	...uploaderProps
+}) {
+	const [itemSelectorPreviewOpen, setItemSelectorPreviewOpen] = useState(
+		false
+	);
+	const [itemSelectorPreviewIndex, setItemSelectorPreviewIndex] = useState(0);
+
+	const itemSelectorPreviewItemsRef = useRef();
+
+	const handleSelectedItem = useCallback(
+		({returnType, value}) => {
+			Liferay.Util.getOpener().Liferay.fire(itemSelectedEventName, {
+				returnType,
+				value,
+			});
+		},
+		[itemSelectedEventName]
+	);
+
+	useEffect(() => {
+		const eventListeners = [];
+
+		eventListeners.push(
+			delegate(
+				document.querySelector(rootNode),
+				'click',
+				'.item-preview',
+				(event) => {
+					handleSelectedItem(event.delegateTarget.dataset);
+				}
+			)
+		);
+
+		const itemSelectorPreviewItems = Array.from(
+			document.querySelectorAll(
+				`#p_p_id${portletNamespace} .item-preview-editable`
+			)
+		).map((node) => node.dataset);
+
+		const clicablePreviewButtons = Array.from(
+			document.querySelectorAll(`#p_p_id${portletNamespace} .icon-view`)
+		);
+
+		if (
```

Comment:
Hey @boton, do you know why this comparison (`.item-preview-editable` vs .`icon-view` elements is necessary? I know it is in the old code as well, but I can't see why we may need this...

Reply:
Isn't it necessary for me I only left it because it is on the old versions, it is a checking that we have the same number of data that trigger elements.